paymentsheet/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -308,3 +319,16 @@ internal sealed interface ScreenState {
 }
 
 internal class NoArgsException : IllegalArgumentException("NativeLinkArgs not found")
+
+private val NativeLinkArgs.configurationWithUpdatedCustomerInfo: LinkConfiguration
+    // The user might have logged out on a previous Link session. We set the email to null
+    // to avoid us defaulting back to the account that they previously logged out of.
+    get() {
+        val clearCustomerDetails = linkAccount == null
+        val effectiveCustomerInfo = if (clearCustomerDetails) {
+            LinkConfiguration.CustomerInfo(null, null, null, null)
```

Comment:
I was thinking of only setting the email address to `null` and keeping the other values as-is, but Web seems to fully clear the customer info for the same flow.